### DISCUSSION TITLE
does "openssl pkcs12" requries FIPS mode or openssl.cnf file in openssl3?

### DISCUSSION THREAD

**Discussion by reguripradeep** [openssl/openssl#23716] [Q&A] [answered]
i am upgrading my openssl from 1.0.2 to 3.2.0. earlier i used to run "openssl pkcs12" after setting OPENSSL_FIPS=1. but no in Openssl3 i dont see any such option. so in openssl3 how can i tell the openssl that  wnt to run the commnd in a FIPS environment?
Also, is pkcs12 not FIPS certified ? i m getting the following error with 3.2.0 as core library and 3.0.8 FIPS provider.
C:\SSL_ossl3\bin>openssl pkcs12 -in server.pfx -out pfxout
Enter Import Password:
Error verifying PKCS12 MAC; no PKCS12KDF support.
Use -nomacver if MAC verification is not required.
i can add "nomacver " but that is not secured right?

**Comment by StephenWall** [comment, marked as answer]
`OPENSSL_FIPS` is only for OpenSSL 1.0.2, it does nothing for OpenSSL 3.  See https://www.openssl.org/docs/man3.0/man7/fips_module.html

**Reply by reguripradeep** [reply]
yes , i even searched in the code and confirmed
